llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -12778,6 +12779,197 @@ struct AAAddressSpaceCallSiteArgument final : AAAddressSpaceImpl {
 };
 } // namespace
 
+/// ------------------------ No Alias Address Space  ---------------------------
+namespace {
```

Comment:
We need document. The most important assumption this AA makes is, the flat AS aliases all others.